rest/src/connection/connectionService.js

```diff
@@ -65,15 +68,31 @@ module.exports.createConnectionService = (config, connectionFactory, authPromise
 				reject(errors.createServiceUnavailableError('connection failed'));
 			});
 
-		return authPromiseFactory(serverSocket, clientKeyPair, apiNodePublicKey, logger)
+		const authPromise = authPromiseFactory(serverSocket, clientKeyPair, apiNodePublicKey, logger)
 			.then(() => {
 				// wrap the socket in a catapult connection and save it
 				const serverConnection = catapultConnection.wrap(serverSocket);
-				if (isPersistent)
+				if (isPersistent) {
 					aliveConnections[node] = serverConnection;
+					delete authenticatingConnectionPromises[node];
+				}
 
+				// return and additionally, return the connection for possible queued connections on `authenticatingConnectionPromises`
 				resolve(serverConnection);
-			}, reject);
+				return serverConnection;
+			}).catch(err => {
+				if (isPersistent)
+					delete authenticatingConnectionPromises[node];
+
+				// reject and additionally, return the connection for possible queued connections on `authenticatingConnectionPromises`
+				reject(err);
+				throw err;
+			});
```

Comment:
not using `finally` for
```
if (isPersistent)
        delete authenticatingConnectionPromises[node];
```

because the `resolve()` code is executed before the `finally()`s which can lead into conflicts